### PR TITLE
feat: hybrid commands

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

File: DisCatSharp.HybridCommands/HybridCommandsExtension.cs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+	internal List<HybridCommandAttribute> registeredCommands { get; set; } = new();
+
+	public static bool DebugEnabled 
+		=> Configuration?.DebugEnabled ?? false;
+
+	internal protected override async void Setup(DiscordClient client)
+	{
+		if (this.Client != null)
+			throw new InvalidOperationException("What did I tell you?");
```
Comment (xMaxximum) [inline DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: what is this error supposed to mean?

**Review by xMaxximum — DISMISSED** [review]

File: DisCatSharp.HybridCommands/HybridCommandsExtension.cs (hunk 1 above)
Comment (Lulalaby) [reply DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: it's just missing all docs...... @TheXorog -.-

Comment (Lulalaby) [reply DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: /// <summary>
	/// Runs setup.
	/// <note type="caution">DO NOT RUN THIS MANUALLY. DO NOT DO ANYTHING WITH THIS.</note>
	/// </summary>
	/// <param name="client">The client to setup on.</param>

Comment (TheXorog) [reply DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: oh nooooo its missing documentation on an internal method we know what it does, oh so terrible

Comment (TheXorog) [reply DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: Docs are not yet a priority, the setup method i copied from another project and modified it to suit my needs

Comment (Lulalaby) [reply DisCatSharp.HybridCommands/HybridCommandsExtension.cs:57]: > oh nooooo its missing documentation on an internal method we know what it does, oh so terrible

Mira. I said it often enough that I aim to document everything so others can work on it as well.
Docs have the same prio as the implementation.

**Pull request closed without merging**